pkgs/applications/terminal-emulators/kitty/default.nix

```diff
@@ -104,6 +104,9 @@ buildPythonApplication rec {
     # Skip `test_ssh_bootstrap_with_different_launchers` when launcher is `zsh` since it causes:
     # OSError: master_fd is in error condition
     ./disable-test_ssh_bootstrap_with_different_launchers.patch
+
+    # Fix test failures due to using global zsh config in build env
+    ./0001-Set-IN_NIX_SHELL-1-for-tests.patch
```

Comment:
How about set `export IN_NIX_SHELL=1` in `preCheck` instead?
We would like to avoid a patch if possible, as it is more difficult to maintain.

Reply:
@natsukium it was my first thought too, but sadly, their test suite overrides external variables completely, so it's impossible to append from outside: https://github.com/kovidgoyal/kitty/blob/913ce58fe3148a71343b1fdda9999657edb2b27e/kitty_tests/shell_integration.py#L35

Reply:
Side note: https://github.com/LnL7/nix-darwin/pull/1051 already merged, which should fix the build issue for unsandboxed (read: Darwin) environments. If carrying a patch is a hassle, we can as well ask users to pick up the `nix-darwin` fix instead. This will mostly address the immediate problem, with a caveat that a different module that pollutes `zshenv` in a similar fashion may again break the build for folks on Darwin. We may address this when it happens though.

Let me know if you'd like to drop the patch and just assume `nix-darwin` users pull the fix.

Reply:
I dropped the IN_NIX_SHELL patch since it's no longer needed with the latest nix-darwin updates; and since carrying a patch complicates packaging.